tesseract/integration-test/src/lib.rs

```diff
@@ -0,0 +1,457 @@
+#![cfg(test)]
+#![deny(missing_docs, unused_imports)]
+
+use anyhow::anyhow;
+use futures::StreamExt;
+use ismp::{
+	consensus::StateMachineHeight,
+	events::{Event, StateMachineUpdated},
+	host::StateMachine,
+	messaging::{Message, Proof, ResponseMessage},
+	router::{Request, RequestResponse},
+};
+
+use pallet_hyperbridge::VersionedHostParams;
+use pallet_ismp_host_executive::HostParam;
+use sc_service::TaskManager;
+use std::{
+	collections::{BTreeMap, HashMap},
+	sync::{Arc, Once},
+};
+use substrate_state_machine::{HashAlgorithm, StateMachineProof, SubstrateStateProof};
+use subxt::{
+	ext::{
+		codec::{Decode, Encode},
+		sp_core::{sr25519::Pair, Pair as PairT},
+	},
+	tx::PairSigner,
+	utils::AccountId32,
+};
+use subxt_signer::sr25519::dev::{self};
+use subxt_utils::{
+	gargantua::{
+		api,
+		api::{
+			ismp::events::{PostRequestHandled, Request as RequestEvent},
+			ismp_demo::events::GetResponse,
+			runtime_types::{
+				ismp::host::StateMachine as StateMachineType,
+				pallet_ismp_demo::pallet::{GetRequest, TransferParams},
+			},
+		},
+	},
+	Hyperbridge,
+};
+use tesseract::logging::setup as log_setup;
+use tesseract_messaging::relay;
+use tesseract_primitives::{config::RelayerConfig, IsmpProvider};
+use tesseract_substrate::{config::KeccakSubstrateChain, SubstrateClient, SubstrateConfig};
+use transaction_fees::TransactionPayment;
+
+// A static `Once` to ensure the initialization code runs only once
+// static INIT: Once = Once::new();
+//
+// // A static `OnceLock` to hold the result of the initialization
+// static INIT_RESULT: OnceLock<Result<(), anyhow::Error>> = OnceLock::new();
+
+static INIT: Once = Once::new();
+
+fn initialize_tracing() -> Result<(), anyhow::Error> {
```

Comment:
This not necessary, you can call `log_setup()` and let it fail silently if the logging has already being initialized.

Reply:
but isnt this much cleaner , as we only allow it to run once and not just let it fail?

Reply:
We don't need the extra code, besides we are sure then log will always be initialized atleast once.

Reply:
done